mitodl/open-discussions
# Use MM and PE data to tag edx courses with program_name and program_type

This PR needs to wait to be merged until after https://github.com/mitodl/micromasters/pull/4317 is merged and deployed.

#### What are the relevant tickets?
#1979 #1981 

#### What's this PR do?
This PR introduces `program_type` and `program_name` fields to the course_catalog Course model. These fields are populated using MicroMasters and Professional Education course data, so that courses in each of these programs are appropriately marked. This PR includes a `task_helpers` function which is called after an mitx sync. This function reads in the MM and PE data, and uses it to tag courses.

#### How should this be manually tested?
Open a python shell and run `tag_edx_course_program`. Check and see if courses were properly tagged with `program_type` and `program_name` fields. The new setting `MICROMASTERS_BASE_URL` should be manually set to `https://micromasters-rc.odl.mit.edu/api/v0/courseruns/` until the current RC is approved and the code deployed to production.
